internal/rbd/replicationcontrollerserver.go

```diff
@@ -108,6 +122,56 @@ func getMirroringMode(ctx context.Context, parameters map[string]string) (librbd
 	return mirroringMode, nil
 }
 
+// getSchedulingDetails gets the mirroring mode and scheduling details from the
+// input GRPC request parameters and validates the scheduling is only supported
+// for mirroring mode.
+func getSchedulingDetails(parameters map[string]string) (admin.Interval, admin.StartTime, error) {
+	admInt := admin.NoInterval
+	adminStartTime := admin.NoStartTime
+	var err error
+
+	val := parameters[imageMirroringKey]
+
+	switch imageMirroringMode(val) {
+	case imageMirrorModeSnapshot:
+	default:
+		return admInt, adminStartTime, status.Error(codes.InvalidArgument, "scheduling is only supported for snapshot mode")
+	}
+
+	// validate mandatory interval field
+	interval, ok := parameters[schedulingIntervalKey]
+	if ok && interval == "" {
+		return admInt, adminStartTime, status.Error(codes.InvalidArgument, "scheduling interval cannot be empty")
+	}
+	adminStartTime = admin.StartTime(parameters[schedulingStartTimeKey])
+	if !ok {
+		// startTime is alone not supported it has to be present with interval
+		if adminStartTime != "" {
+			return admInt, admin.NoStartTime, status.Errorf(codes.InvalidArgument,
+				"%q parameter is supported only with %q",
+				schedulingStartTimeKey,
+				schedulingIntervalKey)
+		}
+	}
+	if interval != "" {
+		admInt, err = validateSchedulingInterval(interval)
+		if err != nil {
+			return admInt, admin.NoStartTime, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+	return admInt, adminStartTime, nil
+}
+
+// validateSchedulingInterval return the interval as it is if its ending with
+// `m|h|d` or else it will return error.
+func validateSchedulingInterval(interval string) (admin.Interval, error) {
+	var re = regexp.MustCompile(`^[0-9]+[mhd]$`)
```

Comment:
This validation fails if we just `d` or `m` or `h`, do we need to treat this as `1d /1m/1h` ?

Reply:
giving only `d` `m` or `h` is not supported and it will fail at the ceph level. it should with a number and suffix.

Reply:
but this is an input we parse and sent to ceph, isnt it @Madhu-1 ?

Reply:
we don't parse(alter)anything from user requests. we just do validation and will send it to ceph

Reply:
@Madhu-1 by `parse` I meant `validation` , also the check here was that, do we need to pass `d/m/h` as `1d/1m/1h` to ceph ? if we think its not required, thats ok.. I was placing a thought here. I am good with any model.

Reply:
IMO let the user pass the format what we expect or else it's going to fail and he can correct it later. @nixpanic any suggestion here?

Reply:
also, not sure, we have to be flexible here for the user to pass `m or M`, `h or H`..etc

Reply:
this is as per the ceph documentation https://docs.ceph.com/en/latest/rbd/rbd-mirroring/#create-image-mirror-snapshots. IMO lets not open up more options and do the internal conversion. if this is really needed we can do as a follow up